### PR TITLE
Add basic readme, examples, build/contributing instructions

### PR DESCRIPTION
**Pull request by TmLev**

### CODE REVIEW

File: docs/cmake.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# CMake
+
+> There are only two kinds of build systems: the ones people complain about and the ones nobody uses.  
+> ---  [lt_algorithm_gt](https://www.reddit.com/r/cpp/comments/ihi37f/what_is_the_point_of_this_talk_just_making_fun_of/g34ex3x?utm_source=share&utm_medium=web2x&context=3)
+> (adapted from Bjarne Stroustrup)
+
+Note, that building tests and benchmarks for __Skipper__ is enabled by default 
+(and for that, `Catch2` and `Google/Benchmark` are required).  
+In order to turn them off, set corresponding variables to `OFF`:
```
Comment (belyakov-am) [inline docs/cmake.md:9]: Add path to that CMake file for more clarity.

File: docs/examples.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ public:
+  // O(1) complexity
+  auto operator*() const -> const T&;
+  auto operator->() const -> const T*;
+  auto operator++(/* prefix */) -> Iterator&;
+  auto operator++(int /* postfix */) -> Iterator;
+  auto operator==(const Iterator& other) const -> bool;
+  auto operator!=(const Iterator& other) const -> bool;
+};
+```
```
Comment (belyakov-am) [inline docs/examples.md:60]: Add `SequentialSkipListMap` Iterator description as some operators (`operator*()` and `operator->()`) are slightly different.

File: docs/cmake.md (hunk 1 above)
Comment (TmLev) [reply docs/cmake.md:9]: It's not necessary for projects that use `skipper` as a library. They should write this in their own `CMakeLists.txt`, not in ours.

**Review by belyakov-am — COMMENTED** [review]
Great job, everything looks so much cool!

File: docs/examples.md (hunk 2 above)
Comment (TmLev) [reply docs/examples.md]: Sorry, fixed.

**Pull request merged**